OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -185,8 +185,9 @@ public override Activity Tick(Actor self)
 					mobile.Facing,
 					mobile.Facing,
 					0);
+				ChildActivity = move;
 
-				return move;
+				return ChildActivity;
```

Comment:
This line goes against the whole idea of child activities, and is also the reason why the change to `MovePart.Queue` below is needed in the first place. You should be able to use `ActivityUtils.RunActivity(self, ChildActivity)` somewhere in this `Tick` method instead. Changing that will also prevent the `MovePart`s from becoming the actor's `CurrentActivity`.